gammapy/obs/observation.py

```diff
@@ -319,3 +339,569 @@ def select_observations(self, selection=None):
+                    and split_name_min[0] == split_name_max[0]):
+                    min_values = axes[i_col].bins
+                    max_values = axes[i_col + 1].bins
+                    edges = np.unique(np.append(min_values, max_values))
+                    # recover units
+                    edges = recover_units(edges, min_values)
+
+                    axes[i_col] = ObservationGroupAxis(split_name_min[0], edges,
+                                                       'bin_edges')
+                    axes.pop(i_col + 1) # remove next entry on the list
+            except:
+                pass
+
+        return axes
+
+    @classmethod
+    def read(cls, filename):
+        """
+        Read observation group definitions from ECSV file.
+
+        Using `~astropy.table.Table` and `~astropy.io.ascii`.
+
+        Parameters
+        ----------
+        filename : str
+            Name of the file.
+
+        Returns
+        -------
+        obs_groups : `~gammapy.obs.ObservationGroups`
+            Observation groups object.
+        """
+        cls.obs_groups_table = ascii.read(filename)
+        return cls(obs_group_axes=cls.table_to_axes(cls, cls.obs_groups_table))
+
+    def write(self, outfile, overwrite=False):
+        """
+        Write observation group definitions to ECSV file.
+
+        Using `~astropy.table.Table` and `~astropy.io.ascii`.
+
+        Parameters
+        ----------
+        outfile : str
+            Name of the file.
+        overwrite : bool, optional
+            Flag to control file overwriting.
+        """
+        # there is no overwrite option in `~astropy.io.ascii`
+        if not os.path.isfile(outfile) or overwrite:
+            ascii.write(self.obs_groups_table, outfile,
+                        format='ecsv', fast_writer=False)
+
+    @property
+    def info(self):
+        """Info string (str)"""
+        s = ''
+        # loop over observation axes
+        for i_axis in np.arange(len(self.obs_group_axes)):
+            s += self.obs_group_axes[i_axis].info
+            if i_axis < len(self.obs_group_axes) - 1:
+                s += '\n'
+        return s
+
+    def info_group(self, group_id):
+        """Group info string
+
+        Parameters
+        ----------
+        group_id : int
+            ID of the group to gather info on.
+
+        Returns
+        -------
+        s : str
+            Group info string.
+        """
+        s = 'group {}:'.format(group_id)
+        # find group row in obs groups table
+        group_ids = self.obs_groups_table['GROUP_ID'].data
+        group_index = np.where(group_ids==group_id)
+        row = group_index[0][0]
+        # loop over observation axes
+        for i_axis in np.arange(len(self.obs_group_axes)):
+            if i_axis != 0:
+                s +=','
+            s += ' ' + self.obs_group_axes[i_axis].name + ' = '
+            if self.obs_group_axes[i_axis].format == 'bin_edges':
+                s += '[' +\
```

Comment:
Such multi-line expressions are very hard to read (and debug if there's an issue).
Please split it in separate statements:
```python
s += ...
s += ...
s += ...
```